### DISCUSSION TITLE
year_vtg and year_act

### DISCUSSION THREAD

**Discussion by shreeyashn20** [iiasa/message_ix#741] [Q&A] [answered]
I am using the MESSAGEix framework to build a model. My model works only when year_vtg is equal to year_act. Following are two examples 

Example : Not Working
```
Technology	year_vtg	year_act
Grid	                2013	2021
Grid	                2013	2022
Grid	                2013	2023
```

Example : Working
```
Technology	year_vtg	year_act
Grid	                2021	2021
Grid	                2022	2022
Grid	                2023	2023
```

The Westeros tutorial uses `vintage_and_active_years() ` function which gives pair of same year_vtg and year_act. 

Now, The Grid is already installed in 2013. Why would I install new grid every year ? Please explain. 

P.S. I am using MESSAGEix 3.4.0

**Comment by khaeru** [comment, marked as answer]
The key concept at the root of your question seems to be similar to the one in #679.

- `year_vtg`, "vintage year/period", or $y^V$ is the period in which capacity (`CAP`) of a particular technology is constructed or created (`CAP_NEW`). If it helps, think of a single substation in the electrical grid.
- `year_act`, "active year/period", or $y^A$ is the period in which the technology operates (`ACT`).
- As you note, a technology constructed in $y^V = 2021$ can be active in $y^A \in (2021, 2022, 2023, ... )$—which periods depends on its `technical_lifetime` and the `duration_period` of those periods.
- Equally, a technology constructed in $y^V = 2013$ could be active in $y^A \in (..., 2021, 2022, 2023, ...)$.
- A technology constructed in $y^V = 2023$ can never be active in $y^A = 2020$; this doesn't make any sense.

In MESSAGE, both `ACT` and `CAP` are *decision variables*: the solver chooses an optimal value of each one. The solver does this for only the periods $y$ (thus $y^V$ and $y^A$) that are in the **model horizon** or optimization years. A MESSAGE Scenario can contain historical periods *before* the model horizon. For instance, our global model contains historical periods like 1950, 1960, …, 2010, 2015; but the "first model year" or $y_0$ is 2020. So the solver **can** choose `CAP` and `ACT` values for $y_0 \le y$, in other words $y \in (2020, 2025, 2030, ...)$ but it **cannot** choose $CAP > 0$ for $y^V = 2015$, because this period is *outside the model's horizon*.

You haven't given a complete example, so I can only guess. But perhaps in the model you have constructed, the year 2013 is not part of the model horizon. If you give technology parameters only for $t=\text{“grid”}, y^V = 2013$, then the solver can never choose to construct this technology, and there will be no way to meet demand.

If that's correct, you could try things like:
- Adjust the model horizon, for instance by changing the first model year.
- Add values to `historical_new_capacity` to tell the model how much `CAP` was constructed in the historical periods before $y_0$.
